app/uk/gov/hmrc/tai/connectors/HttpHandler.scala

```diff
@@ -16,103 +16,115 @@
-            Future.failed(new HttpException(httpResponse.body, httpResponse.status))
+            throw new HttpException(response.body, response.status)
         }
+      }
     }
-  }
 
-  def putToApi[I](url: String, data: I)(implicit hc: HeaderCarrier, rds: HttpReads[I], writes: Writes[I]): Future[HttpResponse] = {
-    http.PUT[I, HttpResponse](url, data).flatMap { httpResponse =>
-      httpResponse.status match {
-
-        case OK =>
-          Future.successful(httpResponse)
-
-        case NOT_FOUND =>
-          Logger.warn(s"HttpHandler - No data can be found")
-          Future.failed(new NotFoundException(httpResponse.body))
-
-        case INTERNAL_SERVER_ERROR =>
-          Logger.warn(s"HttpHandler - Internal Server Error received")
-          Future.failed(new InternalServerException(httpResponse.body))
+    http.PUT[I, HttpResponse](url, data)
+  }
 
-        case BAD_REQUEST =>
-          Logger.warn(s"HttpHandler - Bad Request received")
-          Future.failed(new BadRequestException(httpResponse.body))
+  def postToApi[I](url: String, data: I)(implicit hc: HeaderCarrier, writes: Writes[I]): Future[HttpResponse] = {
 
-        case _ =>
-          Logger.warn(s"HttpHandler - Server error received")
-          Future.failed(new HttpException(httpResponse.body, httpResponse.status))
+    implicit val rawHttpReads: HttpReads[HttpResponse] = new HttpReads[HttpResponse] {
+      override def read(method: String, url: String, response: HttpResponse): HttpResponse = {
+        response.status match {
+          case OK | CREATED => response
+          case _ =>
+            Logger.warn(s"HttpHandler - Error received with status: ${response.status} and body: ${response.body}")
+            throw new HttpException(response.body, response.status)
+        }
       }
     }
+
+    http.POST[I, HttpResponse](url, data)
   }
 
-  def postToApi[I](url: String, data: I)(implicit hc: HeaderCarrier, rds: HttpReads[I], writes: Writes[I]): Future[HttpResponse] = {
-    http.POST[I, HttpResponse](url, data) flatMap { httpResponse =>
-      httpResponse status match {
-        case OK | CREATED =>
-          Future.successful(httpResponse)
+  def deleteFromApi(url: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
 
-        case _ =>
-          Logger.warn(s"HttpHandler - Error received with status: ${httpResponse.status} and body: ${httpResponse.body}")
-          Future.failed(new HttpException(httpResponse.body, httpResponse.status))
+    implicit val rawHttpReads: HttpReads[HttpResponse] = new HttpReads[HttpResponse] {
+      override def read(method: String, url: String, response: HttpResponse): HttpResponse = {
+        response.status match {
+          case OK | NO_CONTENT | ACCEPTED => response
+          case _ =>
+            Logger.warn(s"HttpHandler - Error received with status: ${response.status} and body: ${response.body}")
+            throw new HttpException(response.body, response.status)
+        }
```

Comment:
Could wrap rawHttpReads content into a common function as it's the same in here and postToApi